src/test/java/org/javarosa/core/model/DynamicSelectUpdateTest.java

```diff
@@ -0,0 +1,323 @@
+    }
+
+    private static XFormsElement getSelectFromRepeatForm() {
+        return getSelectFromRepeatForm("");
+    }
+
+    private static XFormsElement getSelectFromRepeatForm(String predicate) {
+        return html(
+            head(
+                title("Select from repeat"),
+                model(
+                    mainInstance(
+                        t("data id=\"repeat-select\"",
+                            t("repeat",
+                                t("value"),
+                                t("label")),
+                            t("select"),
+                            t("filter"))))),
+            body(
+                repeat("/data/repeat",
+                    input("value"),
+                    input("label")),
+                select1Dynamic("/data/select", "../repeat" + (!predicate.isEmpty() ? "[" + predicate + "]" : "")),
+                input("filter")
+            )
+        );
+    }
+    //endregion
+
+    //region Select in repeat
+    // When a dynamic select is in a repeat, the itemsets for all repeat instances are represented by the same ItemsetBinding.
+    @Test
+    public void selectInRepeat_withPredicate_evaluatesChoiceListForEachRepeatInstance() throws Exception {
+        Scenario scenario = Scenario.init("Select from repeat", html(
+            head(
+                title("Select in repeat"),
+                model(
+                    mainInstance(
+                        t("data id=\"repeat-select\"",
+                            t("repeat",
+                                t("select"),
+                                t("filter")))),
+
+                    instance("choices",
+                        item("a", "A"),
+                        item("aa", "AA"),
+                        item("b", "B"),
+                        item("bb", "BB"))),
+                body(
+                    repeat("/data/repeat",
+                        select1Dynamic("/data/repeat/select", "instance('choices')/root/item[starts-with(value,current()/../filter)]"),
+                        input("filter"))
+                )
+            )));
+
+        scenario.answer("/data/repeat[0]/filter", "a");
+        scenario.answer("/data/repeat[1]/filter", "a");
+        assertThat(scenario.choicesOf("/data/repeat[0]/select").size(), is(2));
+        assertThat(scenario.choicesOf("/data/repeat[1]/select").size(), is(2));
+
+        scenario.answer("/data/repeat[1]/filter", "bb");
+        assertThat(scenario.choicesOf("/data/repeat[0]/select").size(), is(2));
+        assertThat(scenario.choicesOf("/data/repeat[1]/select").size(), is(1));
+    }
+    //endregion
+
+    //region Multi-language
+    @Test
+    public void multilanguage() throws Exception {
+        Scenario scenario = Scenario.init("Select from repeat", html(
+            head(
+                title("Multilingual dynamic select"),
+                model(
+                    t("itext",
+                        t("translation lang=\"fr\"",
```

Comment:
Any idea why we‘re using `\"` instead of `'`? Both forms seem to generate double-quoted attribute values. I find code with escaped double quotes harder to read.

Reply:
I didn't realize that'd do the right thing. 👍 will update.

Reply:
<img width="975" alt="image" src="https://user-images.githubusercontent.com/55099/109849839-b1d36000-7c06-11eb-82bf-efaaf51779ad.png">

Here’s where the attribute value delimiters are ignored, in XFormsElement. Later they are replaced with double quotes.